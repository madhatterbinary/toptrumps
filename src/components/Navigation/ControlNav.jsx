import React from 'react';
import PropTypes from 'prop-types';
import Button from '../common/Button';
import { STATUS_READY, STATUS_OVER } from '../../constants/constants';

const ControlNav = (props) => { 
  const {
    status,
    isPlayEnabled,
    handleButtonClick,
  } = props;

  let buttonLabel;
  switch(status) {
    case STATUS_READY:
      buttonLabel = 'Play';
      break;
    case STATUS_OVER:
      buttonLabel = 'Play Again';
      break;
    default:
      buttonLabel = 'Shuffle';
  }

  return (
    <div className="control-nav">
      <Button
        buttonClassName={status === STATUS_READY ? 'dark' : 'top-trump'}
        buttonOnClick={handleButtonClick}
        buttonLabel={buttonLabel}
        buttonIsDisabled={!isPlayEnabled}
      />
    </div>
  );
}

ControlNav.propTypes = {
  status: PropTypes.string.isRequired,
  isPlayEnabled: PropTypes.bool.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};

export default ControlNav;
