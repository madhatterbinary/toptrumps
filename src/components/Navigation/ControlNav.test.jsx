import React from 'react';
import ControlNav from './ControlNav';
import TestRenderer from 'react-test-renderer';

describe('ControlNav', () => {
  it('renders', () => {
    const component = TestRenderer.create(<ControlNav status="" handleButtonClick={()=>{}} isPlayEnabled={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
