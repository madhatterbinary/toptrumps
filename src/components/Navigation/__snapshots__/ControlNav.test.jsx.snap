// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlNav renders 1`] = `
<div
  className="control-nav"
>
  <button
    className="btn top-trump disabled"
    disabled={true}
    onClick={[Function]}
  >
    Shuffle
  </button>
</div>
`;
